nengo/utils/compat.py

```diff
@@ -75,12 +75,24 @@ def ensure_bytes(s):
 assert TextIO
 
 
+def as_ascii(string):
+    if not PY2 and isinstance(string, bytes):  # Python 3
```

Comment:
Is the `isinstance` check here necessary? By default, I believe, the `str` type in Python 3 is unicode, so those would not be turned to ascii... I'll write a few tests for this, perhaps.